ipn/ipnlocal/local.go

```diff
@@ -219,6 +219,14 @@ type LocalBackend struct {
 	serveListeners     map[netip.AddrPort]*serveListener // addrPort => serveListener
 	serveProxyHandlers sync.Map                          // string (HTTPHandler.Proxy) => *httputil.ReverseProxy
 
+	// getTCPHandlerForFunnelFlow conditionally handles an incoming TCP flow for
```

Comment:
looks like this is protected by the mutex on a paragraph of fields above. does that mutex protect all these other fields too (like `serveProxyHandlers`)? that would be a little surprising. assuming it doesn't, I think this field should be mashed into the very large paragraph with mu. (might be time for a new private type or something for those grouped fields just for readability)

Reply:
It didn't need a mutex, I think I just had it a remnant of a previous iteration of the logic. Moved and updated docs